data-serving/data-service/src/model/vaccine.ts

```diff
@@ -0,0 +1,23 @@
+import mongoose from 'mongoose';
+import { symptomsSchema, SymptomsDocument } from './symptoms';
+
+export const vaccineSchema = new mongoose.Schema(
+    {
+        name: String,
+        batch: String,
+        date: Date,
+        sideEffects: [symptomsSchema],
+        previousInfection: String,
+        previousInfectionDetectionMethod: String,
+    },
+    { _id: false },
+);
+
+export type VaccineDocument = mongoose.Document & {
+    name: string;
+    batch: string;
+    date: Date;
+    sideEffects: [SymptomsDocument];
+    previousInfection: 'yes' | 'no' | 'NA';
```

Comment:
Could this be boolean? That way presence of previousInfection would have true / false and absence indicates no previous infection or data not known.

Reply:
The request at #2120 was for yes/no/NA. I guess we have to ask @mougk and other experts what the expectation is.